docs/src/main/java/org/geotools/tutorial/coverage/ImageTiler.java

```diff
@@ -0,0 +1,224 @@
+        double coverageMaxX = coverageEnvelope.getBounds().getMaxX();
+        double coverageMinY = coverageEnvelope.getBounds().getMinY();
+        double coverageMaxY = coverageEnvelope.getBounds().getMaxY();
+
+        int htc = this.getNumberOfHorizontalTiles() != null
+                ? this.getNumberOfHorizontalTiles() : NUM_HORIZONTAL_TILES;
+        int vtc = this.getNumberOfVerticalTiles() != null
+                ? this.getNumberOfVerticalTiles() : NUM_VERTICAL_TILES;
+
+        double geographicTileWidth = (coverageMaxX - coverageMinX) / htc;
+        double geographicTileHeight = (coverageMaxY - coverageMinY) / vtc;
+
+        CoordinateReferenceSystem targetCRS = gridCoverage.getCoordinateReferenceSystem();
+        //docs end envelope
+
+        //make sure to create our output directory if it doesn't already exist
+        File tileDirectory = this.getOutputDirectory();
+        if (!tileDirectory.exists()) {
+            tileDirectory.mkdirs();
+        }
+
+        //iterate over our tile counts
+        for (int i = 0; i < htc; i++) {
+            for (int j = 0; j < vtc; j++) {
+
+                //create the envelope of the tile
+                double envelopeStartX = (i * geographicTileWidth) + coverageMinX;
+                double envelopeEndX = envelopeStartX + geographicTileWidth;
+                double envelopeStartY = (j * geographicTileHeight) + coverageMinY;
+                double envelopeEndY = envelopeStartY + geographicTileHeight;
+
+                Envelope envelope = new ReferencedEnvelope(
+                        envelopeStartX, envelopeEndX, envelopeStartY, envelopeEndY, targetCRS);
+
+                CoverageProcessor processor =  CoverageProcessor.getInstance();
+
+                //docs start cropping
+                //An example of manually creating the operation and parameters we want
+                final ParameterValueGroup param = processor.getOperation("CoverageCrop").getParameters();
+                param.parameter("Source").setValue(gridCoverage);
+                param.parameter("Envelope").setValue(envelope);
+
+                GridCoverage2D finalCoverage = (GridCoverage2D) processor.doOperation(param);
+                //docs end cropping
+                //docs start scale
+                if (this.getTileScale() != null) {
+                    //An alternative to using parameters to do the operations, we can use the
+                    //Operations class to do them in a slightly more type safe way.
+                    Operations ops = new Operations(null);
+                    finalCoverage = (GridCoverage2D) ops.scale(
+                            finalCoverage, this.getTileScale(), this.getTileScale(), 0, 0);
+                }
+
+                //docs end scale
+
+                //docs start end
+                File tileFile = new File(tileDirectory, i + "_" + j + "." + fileExtension);
+                format.getWriter(tileFile).write(finalCoverage, null);
```

Comment:
Same deal here for this block of code - factor out a method so we can javadoc it and take it as a separate step.

```
writeTile( source, bounds, tileFile )
```

And I *love* showing both CoverageProcessor and Operations - this will be a great tutorial.